monero/wallet_client.go

```diff
@@ -221,49 +250,211 @@ func (c *walletClient) WaitForReceipt(req *WaitForReceiptRequest) (*wallet.Trans
 }
 
 func (c *walletClient) Transfer(
+	ctx context.Context,
 	to mcrypto.Address,
 	accountIdx uint64,
 	amount *coins.PiconeroAmount,
-) (*wallet.TransferResponse, error) {
+	numConfirmations uint64,
+) (*wallet.Transfer, error) {
 	amt, err := amount.Uint64()
 	if err != nil {
 		return nil, err
 	}
-	return c.wRPC.Transfer(&wallet.TransferRequest{
+	amountStr := amount.AsMoneroString()
+	log.Infof("Transferring %s XMR to %s", amountStr, to)
+	reqResp, err := c.wRPC.Transfer(&wallet.TransferRequest{
 		Destinations: []wallet.Destination{{
 			Amount:  amt,
 			Address: string(to),
 		}},
 		AccountIndex: accountIdx,
 	})
+	if err != nil {
+		log.Warnf("Transfer of %s XMR failed: %s", amountStr, err)
+		return nil, fmt.Errorf("transfer failed: %w", err)
+	}
+	log.Infof("Transfer of %s XMR initiated, TXID=%s", amountStr, reqResp.TxHash)
+	transfer, err := c.waitForReceipt(&waitForReceiptRequest{
+		Ctx:              ctx,
+		TxID:             reqResp.TxHash,
+		NumConfirmations: numConfirmations,
+		AccountIdx:       accountIdx,
+	})
+	log.Warnf("Transfer TXID=%s failed to get receipt: %s", reqResp.TxHash, err)
```

Comment:
do we need a warn here since we also return an error?

Reply:
also this should be in the `if err != nil` block

Reply:
@dimalinux this shouldn't be logging here?

Reply:
Those changes were pushed before well before this thread was resolved the first time:
https://github.com/AthanorLabs/atomic-swap/commit/91af4ae32337f1d28d79540b93bcb161c07c9325